test-lit/test_splitter.py

```diff
@@ -20,22 +19,25 @@ def __init__(self):
         self.modfile_tests = []
         self.symbol_tests = []
         self.generic_tests = []
+        self.folding_tests = []
         self.unsupported_tests = []
-        self.test_types = ['ERROR', 'MODFILE', 'SYMBOL', 'GENERIC']
-        root = os.getcwd()
-        while not root.endswith("f18"):
-            root = os.path.abspath(os.path.dirname(root))
+        self.test_types = ['ERROR', 'MODFILE', 'SYMBOL', 'GENERIC', 'FOLDING']
+        root = Path(__file__).resolve()
+        while root.name != "f18":
+            root = root.parent
         self.root = root
-        self.test_path = os.path.join(self.root, "test")
+        self.test_path = self.root / "test"
```

Comment:
It's not the most important thing in the world, but using os.path.join is always better than hard coding path separators like this. I really don't understand why you have gone backwards from this. What's the rationale?

Reply:
Unifying the use of Pathlib over os.path across the files. The alternative to this is self.root.joinpath("test") so I'll switch to that if it looks nicer.

Reply:
Yes please